go-controller/cmd/ovn-kube-cni-overlay/ovn-kube-cni-overlay.go

```diff
@@ -217,30 +227,39 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, string(out))
 	}
 
-	// TODO: conform with cni specification
-	result := fmt.Sprintf("{\"ip_address\":\"%s\", \"mac_address\":\"%s\", \"gateway_ip\": \"%s\"}", ipAddress, macAddress, gatewayIP)
-	_, err = os.Stdout.Write([]byte(result))
-
-	return err
-}
-
-func cmdDel(args *skel.CmdArgs) error {
-	argsMap, err := argString2Map(args.Args)
+	// Build the result structure to pass back to the runtime
+	addr, addrNet, err := net.ParseCIDR(ipAddress)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to parse IP address %q: %v", ipAddress, err)
 	}
-	containerID := argsMap["K8S_POD_INFRA_CONTAINER_ID"]
-	if containerID == "" {
-		return fmt.Errorf("required CNI variable missing")
+	ipVersion := "6"
+	if addr.To4() != nil {
+		ipVersion = "4"
+	}
+	result := &current.Result{
+		Interfaces: []*current.Interface{hostIface, contIface},
+		IPs: []*current.IPConfig{
+			{
+				Version:   ipVersion,
+				Interface: current.Int(1),
```

Comment:
Are we sure about hardcoding this to '1'? I don't have an alternative answer. And I do not see how this is affecting anything, but still.
We have cases of ovn being  used in multi-network scenario.